### PR TITLE
fix: fix build break and upgrade dependencies

### PR DESCRIPTION
**Pull request by raymondfeng**
- fix build break by newer version of prettier (https://travis-ci.org/strongloop/loopback-next/jobs/327503885)
- upgrade tool dependencies

## Checklist

- [x] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] Related API Documentation was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `packages/example-*` were updated

### CODE REVIEW

File: packages/build/config/tsconfig.common.json [hunk 1]
```diff
@@ -5,7 +5,6 @@
     "experimentalDecorators": true,
     "noImplicitAny": true,
     "strictNullChecks": true,
-
```
Comment (bajtos) [inline packages/build/config/tsconfig.common.json]: JSON files should not be linted by prettier. I find this empty line as very useful to delimit language-related flags from project-related settings.

Could you please revert this change?

File: packages/build/config/tslint.common.json [hunk 2]
```diff
@@ -22,7 +22,6 @@
     "no-unused-expression": true,
     "no-unused-variable": true,
     "no-var-keyword": true,
-    "triple-equals": [true, "allow-null-check", "allow-undefined-check"],
-    "typeof-compare": true
```
Comment (bajtos) [inline packages/build/config/tslint.common.json]: Could you please explain why you are removing `typeof-compare`? I am sure you have good reasons, I'd just like to understand them too 😄

Comment (raymondfeng) [reply packages/build/config/tslint.common.json]: Running the upgraded version of `tslint` complains that `typeof-compare` rule is now the default since TS 2.2 and it's not needed.

**Comment by raymondfeng** [comment]
@bajtos PTAL

File: packages/build/config/tslint.common.json (hunk 2 above)
Comment (kjdelisle) [reply packages/build/config/tslint.common.json]: ^^^ It even prints that warning in our CI after tests pass (Travis & AppVeyor)
